### PR TITLE
config: officially deprecating deprecated_v1 fields

### PR DESCRIPTION
**Pull request by alyssawilk**
For removal in the next release ~3 months out

*Risk Level*: Low (deprecating one lingering proto field)
*Testing*: n/a
*Docs Changes*: updated DEPRECATED.md
*Release Notes*: n/a
*Deprecated*: deprecated deprecated config

### CODE REVIEW

File: DEPRECATED.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ A logged warning is expected for each deprecated item that is in deprecation win
 
 ## Version 1.9.0 (pending)
 
+* Use of the v2 API "deprecated_v1" fields is deprecated.
```
Comment (htuch) [inline DEPRECATED.md]: Nit: `*deprecated_v1*`

File: DEPRECATED.md [hunk 2]
```diff
@@ -8,9 +8,11 @@ A logged warning is expected for each deprecated item that is in deprecation win
 
 ## Version 1.9.0 (pending)
 
+* Use of the v2 API *deprecated_v1* fields is deprecated.
+
 ## Version 1.8.0 (Oct 4, 2018)
 
-* Use of the v1 API (including `*.deprecated_v1` fields in the v2 API) is deprecated.
+* Use of the v1 API is deprecated.
```
Comment (htuch) [inline DEPRECATED.md]: No rewriting history! :)

**Comment by rshriram** [comment]
does this include tcp_proxy? coz we couldn't remove the tcp proxy deprecated_v1 config (@venilnoronha had a Pr). I think it boiled down to lack of source IP match in filter chain match.

**Comment by venilnoronha** [comment]
PR #4625 is still open.

File: api/envoy/api/v2/lds.proto [hunk 3]
```diff
@@ -98,7 +98,7 @@ message Listener {
   }
 
   // [#not-implemented-hide:]
-  DeprecatedV1 deprecated_v1 = 7;
+  DeprecatedV1 deprecated_v1 = 7 [deprecated = true];
```
Comment (mattklein123) [inline api/envoy/api/v2/lds.proto]: I think there are more than this as @rshriram points out? I'm pretty sure we have a bunch of these...

Comment (alyssawilk) [reply api/envoy/api/v2/lds.proto]: I believe the rest were already deprecated :-)
I'm going to close this out regardless though - I'd thought updating the deprecation notice from last release to this release would make deprecation schedule more clear but if we'd prefer to not rewrite history I don't think deprecating it every version until we remove is helpful

**Pull request closed without merging**